contracts/modules/TransferManager/BlacklistTransferManager.sol

```diff
@@ -0,0 +1,278 @@
+        bytes32 _name, 
+        uint256 _repeatPeriodTime
+    );
+    
+    // Emit when the added blacklist type is deleted
+    event DeleteBlacklistType(
+        bytes32 _name
+    );
+
+    // Emit when new investor is added to the blacklist type
+    event AddInvestorToBlacklist(
+        address _investor, 
+        bytes32 _blacklistName
+    );
+    
+    // Emit when investor is deleted from the blacklist type
+    event DeleteInvestorFromBlacklist(
+        address _investor,
+        bytes32 _blacklist
+    );
+
+    // Emit when investor is deleted from the blacklist type
+    event LogTransfer(
+        uint256 _endTime,
+        uint256 _startTime,
+        uint256 _repeat,
+        uint256 _blacklist,
+        uint256 _repeatar,
+        uint256 _now
+        
+    );
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+    public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+    * @notice This function returns the signature of configure function
+    */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(0);
+    }
+
+
+    /** 
+    * @notice Used to verify the transfer transaction
+    * @param _from Address of the sender
+    * @dev Restrict the blacklist address to transfer token 
+    * if the current time is in the time frame define for the 
+    * blacklist type associated with the blacklist address
+    */
+    function verifyTransfer(address _from, address /* _to */, uint256 /* _amount */, bytes /* _data */, bool /* _isTransfer */) public returns(Result) {
+        if(!paused){
+            if(investorToBlacklist[_from].length != 0){
+                for(uint256 i = 0; i < investorToBlacklist[_from].length; i++){
+                    if(investorToBlacklist[_from][i] != bytes32(0)){
+                        uint256 endTimeTemp = blacklists[investorToBlacklist[_from][i]].endTime;
+                        uint256 startTimeTemp = blacklists[investorToBlacklist[_from][i]].startTime;
+                        uint256 repeatPeriodTimeTemp = blacklists[investorToBlacklist[_from][i]].repeatPeriodTime * 1 days;
+                        // blacklistTime time is used to find the new startTime and endTime 
+                        // suppose startTime=500,endTime=1500,repeatPeriodTime=500 then blacklistTime =1500
+                        // if you add blacklistTime to startTime and endTime i.e startTime = 2000 and endTime = 3000
+                        uint256 blacklistTime = (endTimeTemp.sub(startTimeTemp)).add(repeatPeriodTimeTemp);
+                        // Find the repeating parameter that will be used to calculate the new startTime and endTime
+                        // based on the new current time value   
+                        uint256 repeater = now.div(endTimeTemp.add(repeatPeriodTimeTemp));
```

Comment:
This doesn't sound right. `repeater` will always be 0 or 1 with this. 
Maybe you want something like` repeater = now.sub(startTimeTemp).div(blacklistTime)`

Reply:
@maxsam4 Done the suggested changes. Thanks for the suggestion.

Reply:
I could not understand above logic to check lock period.
I proposal below logic code.
basically algorithm find nearest start and nearest end with now (but before now). Then compare them.

```
if(investorToBlacklist[_from][i] != bytes32(0)){
                    uint256 endTimeTemp = blacklists[investorToBlacklist[_from][i]].endTime;
                    uint256 startTimeTemp = blacklists[investorToBlacklist[_from][i]].startTime;                    
                    uint256 repeatPeriodTimeTemp = blacklists[investorToBlacklist[_from][i]].repeatPeriodTime * 1 days;
                    
                    // always in locking case
                    if(endTimeTemp.sub(startTimeTemp) < repeatPeriodTimeTemp){
                        return Result.INVALID
                    }

                    uint256 repeaterFromStartToNow = now.sub(startTimeTemp).div(repeatPeriodTimeTemp)
                    uint256 nearestStart = repeaterFromStartToNow.mul(repeatPeriodTimeTemp) + startTimeTemp

                    uint256 repeaterFromEndToNow = now.sub(endTimeTemp).div(repeatPeriodTimeTemp)
                    uint256 nearestEnd = repeaterFromStartToNow.mul(repeatPeriodTimeTemp) + endTimeTemp

                    // in locking period
                    if(nearestStart > nearestEnd){
                        return Result.INVALID
                    }

                    return Result.NA;
                }
```